messages/Cargo.toml

```diff
@@ -16,15 +16,15 @@ test-util = []
 anyhow.workspace = true
 base64.workspace = true
 derivative.workspace = true
-hex = "0.4"
-num_enum = "0.7.2"
+hex = { workspace = true }
+num_enum = { workspace = true }
 # We can't pull prio in from the workspace because that would enable default features, and we do not
 # want prio/crypto-dependencies
-prio = { version = "0.16.2", default-features = false, features = ["multithreaded", "experimental"] }
+prio = { workspace = true, features = ["multithreaded", "experimental"] }
```

Comment:
We should move `default-features = false` up to the workspace level (or roll this line back) to avoid requiring the crypto-dependencies feature from here when not necessary. Mozilla isn't currently using `janus_messages`, but this change would block them from doing so.

Reply:
I think that's insufficient. I tried this patch:

```patch
diff --git a/messages/Cargo.toml b/messages/Cargo.toml
index 8c8515a1..d0c9bdfc 100644
--- a/messages/Cargo.toml
+++ b/messages/Cargo.toml
@@ -20,7 +20,7 @@ hex = { workspace = true }
 num_enum = { workspace = true }
 # We can't pull prio in from the workspace because that would enable default features, and we do not
 # want prio/crypto-dependencies
-prio = { workspace = true, features = ["multithreaded", "experimental"] }
+prio = { workspace = true, default-features = false, features = ["multithreaded", "experimental"] }
 rand.workspace = true
 serde.workspace = true
 thiserror.workspace = true
```

Running `cargo tree -p janus_messages` with this applied, I see the following.

```
...
├── prio v0.16.2
│   ├── aes v0.8.4
...
```

Using this patch instead, we get what we want. The same `cargo tree -p janus_messages` command shows that `prio` is only pulling in `sha3`, not `aes` etc.

```patch
diff --git a/Cargo.toml b/Cargo.toml
index 42c9ff83..986b3f30 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,7 +70,7 @@ opentelemetry_sdk = { version = "0.22", features = ["metrics"] }
 opentelemetry-semantic-conventions = "0.14"
 postgres-protocol = "0.6.6"
 postgres-types = "0.2.6"
-prio = { version = "0.16.2", features = ["multithreaded", "experimental"] }
+prio = { version = "0.16.2", default-features = false, features = ["multithreaded", "experimental"] }
 prometheus = "0.13.3"
 querystring = "1.1.0"
 rand = "0.8"
diff --git a/core/Cargo.toml b/core/Cargo.toml
index 6e822dce..d3e37301 100644
--- a/core/Cargo.toml
+++ b/core/Cargo.toml
@@ -48,7 +48,7 @@ http-api-problem.workspace = true
 janus_messages.workspace = true
 kube = { workspace = true, optional = true, features = ["rustls-tls"] }
 k8s-openapi = { workspace = true, optional = true }
-prio.workspace = true
+prio = { workspace = true, default-features = true }
 rand.workspace = true
 regex = { workspace = true }
 reqwest = { workspace = true, features = ["json"] }
diff --git a/messages/Cargo.toml b/messages/Cargo.toml
index 8c8515a1..2186ecd9 100644
--- a/messages/Cargo.toml
+++ b/messages/Cargo.toml
@@ -18,8 +18,6 @@ base64.workspace = true
 derivative.workspace = true
 hex = { workspace = true }
 num_enum = { workspace = true }
-# We can't pull prio in from the workspace because that would enable default features, and we do not
-# want prio/crypto-dependencies
 prio = { workspace = true, features = ["multithreaded", "experimental"] }
 rand.workspace = true
 serde.workspace = true
```

This is because features selected in package files are additive on top of those selected in the workspace file. If we don't include `default-features = false` at the workspace level, there's no way to go back.